coverage test upload to coveralls broken

It looks like the interface for uploading coverage to coveralls has changed, i.e. it is broken in recent PRs e.g. #1105, even though coveralls service claims everything is up.  This doesn't require long-weekend immediate action, but it would be nice to get it sorted out next week (or otherwise temporarily disable coverage tests if it is tricky to fix).

From the #1105 logs:
```
Submitting coverage to coveralls.io...
11
Could not submit coverage: 422 Client Error: Unprocessable Entity for url: https://coveralls.io/api/v1/jobs
12
resubmitting with id None-21428986068999282
13
Traceback (most recent call last):
14
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/coveralls/api.py", line 252, in wear
15
    response.raise_for_status()
16
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
17
    raise HTTPError(http_error_msg, response=self)
18
requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: https://coveralls.io/api/v1/jobs
19

20
During handling of the above exception, another exception occurred:
21

22
Traceback (most recent call last):
23
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/coveralls/cli.py", line 85, in main
24
    result = coverallz.wear()
25
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/coveralls/api.py", line 255, in wear
26
    raise CoverallsException('Could not submit coverage: {}'.format(e))
27
coveralls.exception.CoverallsException: Could not submit coverage: 422 Client Error: Unprocessable Entity for url: https://coveralls.io/api/v1/jobs
28
Error: Process completed with exit code 1.
Post Checkout code
```

--- a/.github/workflows/python-package.yml
+++ b/.github/workflows/python-package.yml
@@ -74,6 +74,7 @@ jobs:
               run: DESIMODEL=$(pwd) pytest --cov
             - name: Coveralls
               env:
+                COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
                 GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
               run: coveralls
 

